// containers/admin/posts/PostsList.js
import React, {Component} from 'react'
import {Route} from 'react-router'
import {connect} from 'react-redux'
import {Table, Button, Grid, Row, Col, Modal} from 'react-bootstrap';
import PostForm from './PostForm'
import {updatePost} from 'actions/posts'

class PostsList extends Component {

  constructor(props) {
    super(props);
    this.state = {
      modalEdit: {
        Show: false,
        id: ''
      }
    }
  }


  render() {
    const {posts} = this.props

    return (
      <div>
        <Grid>
          <Row className="show-grid">
            <Col xs={5} md={1}/>
            <Col xs={8} md={10}>
              <Table responsive striped hover bordered condensed>
                <thead>
                <tr>
                  <th width={1}>#</th>
                  <th width='100%'>Title</th>
                  <th colSpan='2' width='1'>
                    <Button bsStyle="success">Add</Button>
                  </th>
                </tr>
                </thead>
                <tbody>
                {posts.map((row) => {
                  return (<tr key={row.id.toString()}>
                      <td key={row.id.toString()}>{row.id}</td>
                      <td key={row.title}>{row.title}</td>
                      <td key="{row.id}_edit">
                        <Button
                          bsStyle="warning"
                          onClick={() => this.setState({modalEdit: {Show: true, id: row.id}})}>
                          Edit
                        </Button>
                      </td>
                      <td key="{row.id}"><Button bsStyle="danger">Delete</Button></td>
                    </tr>
                  )
                })}
                </tbody>
              </Table>
            </Col>
            <Col xs={5} md={1}/>
          </Row>
        </Grid>
      </div>
    )
  }
}

export default connect(
  state => ({
    posts: state.posts || []
  }),
  ({
    updatePost: updatePost
  })
)(PostsList)
