// containers/posts/index.js
import React, {Component} from 'react';

import {connect} from 'react-redux';
import {fetchPosts} from 'actions/posts';

class Posts extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  componentWillMount() {
    this.props.fetchPosts();

  }
  //
  // renderPost = (post, index) => {
  //   return (
  //     <div key={index}>
  //       {post.title}
  //     </div>
  //   )
  // }
  //
  // renderPosts = () => {
  //   const {posts} = this.props
  //   // return Object.keys(posts).forEach( key => {
  //   //   this.renderPost(posts[key],key)
  //   // });
  //   console.log(' renderPosts - posts ',posts)
  //   return (
  //     <div>OK!</div>
  //   )
  // }

  render(){
    // debugger
    // const {posts} = this.props
    // console.log('render - posts',posts)
    // return (
    //   <div>
    //     {this.renderPosts()}
    //   </div>
    // )
    console.log('!!!')
    return(
      <div>123</div>)

  }
}

const mapStateToProps = (state) => {
  return {
    posts: state.posts || []
  }
}

const mapDispatchToProps = {
  fetchPosts: fetchPosts,
}

export default connect(mapStateToProps, mapDispatchToProps)(Posts)
