// actions/posts.js
import actionTypes from 'actionTypes'
import {HTTP} from 'helpers/http'
const {
  FETCH_POSTS_START,
  FETCH_POSTS_SUCCESS,
  FETCH_POSTS_FAILURE,
  UPDATE_POST_START,
  UPDATE_POST_SUCCESS,
  UPDATE_POST_FAILURE,

} = actionTypes

export const fetchPosts = () => (dispatch) => {
  dispatch({type: FETCH_POSTS_START})
  return HTTP('posts',{
    method: 'GET',
  }).then(response => {
    dispatch({
      type: FETCH_POSTS_SUCCESS,
      payload: response
    })
    return response
  }).catch(error => {
    dispatch({
      type: FETCH_POSTS_FAILURE,
      payload: error,
      error: true
    })
    return error
  })
}


export const updatePost = (values) => (dispatch) => {
  dispatch({type: UPDATE_POST_START})
  // return HTTP({
  //   method: 'put',
  //   url: `posts/${values.id}`,
  //   data: {title: '11'},
    return HTTP.put(
      `posts/${values.id}`,
      values
    ).then(response => {
    dispatch({
      type: UPDATE_POST_SUCCESS,
      payload: response
    })
    return response
  }).catch(error => {
    dispatch({
      type: UPDATE_POST_FAILURE,
      payload: error,
      error: true
    })
    return error
  })
}
